### PR TITLE
[Mailer][Webhook] Add MailerSend support

### PR DESCRIPTION
**Pull request by doobas**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? |no
| License       | MIT

Support webhooks from [MailerSend](https://www.mailersend.com/blog/monitor-email-events-using-webhooks)
MailerSend developers [documentation](https://developers.mailersend.com/api/v1/webhooks.html#webhooks-overview)

### CODE REVIEW

File: src/Symfony/Component/Mailer/Bridge/MailerSend/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+7.1
+---
+
+* Added support for `RemoteEvent` and `Webhook`
```
Comment (xabbuh) [inline src/Symfony/Component/Mailer/Bridge/MailerSend/CHANGELOG.md]:
```suggestion
* Add support for `RemoteEvent` and `Webhook`
```

**Comment by xabbuh** [comment]
You will also have to add `symfony/webhook` as a dev dependency to the MailerSend bridge's `composer.json` file.

File: src/Symfony/Component/Mailer/Bridge/MailerSend/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 CHANGELOG
 =========
 
+7.1
+---
+
+* Add support for `RemoteEvent` and `Webhook`
```
Comment (OskarStark) [inline src/Symfony/Component/Mailer/Bridge/MailerSend/CHANGELOG.md]:
```suggestion
 * Add support for `RemoteEvent` and `Webhook`
```

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/mailer_webhook.php [hunk 3]
```diff
@@ -31,6 +33,11 @@
             ->args([service('mailer.payload_converter.brevo')])
         ->alias(BrevoRequestParser::class, 'mailer.webhook.request_parser.brevo')
 
+        ->set('mailer.payload_converter.mailersend', MailerSendPayloadConverter::class)
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/mailer_webhook.php:36]: Indention is wrong

**Review by OskarStark — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Component/Mailer/Bridge/MailerSend/Webhook/MailerSendRequestParser.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        return new ChainRequestMatcher([
+            new MethodRequestMatcher('POST'),
+            new IsJsonRequestMatcher(),
+        ]);
+    }
+
+    protected function doParse(Request $request, #[\SensitiveParameter] string $secret): ?AbstractMailerEvent
+    {
+        if ($secret) {
+
```
Comment (fabpot) [inline src/Symfony/Component/Mailer/Bridge/MailerSend/Webhook/MailerSendRequestParser.php:43]: This empty line should be removed

File: src/Symfony/Component/Mailer/Bridge/MailerSend/Webhook/MailerSendRequestParser.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                $request->headers->get('Signature'),
+                $request->getContent(),
+                $secret,
+            );
+        }
+
+        $content = $request->toArray();
+        if (
+            !isset($content['type'], $content['data']['email']['message']['id'], $content['data']['email']['recipient']['email'])
+        ) {
```
Comment (fabpot) [inline src/Symfony/Component/Mailer/Bridge/MailerSend/Webhook/MailerSendRequestParser.php:58]: Should be all on one line

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
As I cannot push upstream, I've finished the work here: #53740

**Pull request closed without merging**